overlord/hookstate/ctlcmd/ctlcmd.go

```diff
@@ -113,47 +113,50 @@ func (f ForbiddenCommandError) Error() string {
 	return f.Message
 }
 
-// ForbiddenCommand contains information about an attempt to use a command in a context where it is not allowed.
-type ForbiddenCommand struct {
-	Uid  uint32
-	Name string
-}
-
-func (f *ForbiddenCommand) Execute(args []string) error {
-	return &ForbiddenCommandError{Message: fmt.Sprintf("cannot use %q with uid %d, try with sudo", f.Name, f.Uid)}
-}
-
 // nonRootAllowed lists the commands that can be performed even when snapctl
 // is invoked not by root.
 var nonRootAllowed = []string{"get", "services", "set-health", "is-connected", "system-mode"}
 
 // Run runs the requested command.
 func Run(context *hookstate.Context, args []string, uid uint32) (stdout, stderr []byte, err error) {
+	if len(args) == 0 {
+		return nil, nil, fmt.Errorf("snapctl cannot run without args")
```

Comment:
This error condition is already checked in api_snapctl.go - runSnapctl - before calling this function, so maybe at this point we should just say "internal error: snapctl cannot run without args"? In any case, it would be nice to add a unit test for such case (maybe in api_snapctl tests as it seems to be missing such test).

Reply:
+1 for the error message change.
I noticed that the check already happened in the daemon package but since it's a different package and this code depends on that condition, I preferred to repeat the check. I added a `TestRunNoArgsFailure` in ctlcmd_test.go but I can add another to api_snapctl_test.go  since they're targeting different code.

Reply:
Yes it's good to check again, having test for both error checks (i.e. adding a test to  api_snapctl_test.go) would be great.